File: server/src/main/java/de/scameronde/chat/businesstypes/ChatCommand.java
package de.scameronde.chat.businesstypes;

public class ChatCommand {
  String command;
  ChatRegistration registration;
  ChatMessage message;

  public ChatCommand() {

  }

  public String getCommand() {
    return command;
  }

  public void setCommand(String command) {
    this.command = command;
  }

  public ChatRegistration getRegistration() {
    return registration;
  }

  public void setRegistration(ChatRegistration registration) {
    this.registration = registration;
  }

  public ChatMessage getChatMessage() {
    return message;
  }

  public void setChatMessage(ChatMessage message) {
    this.message = message;
  }
}

File: client/client/Businesstypes/ChatCommand.hs
{-# LANGUAGE AllowAmbiguousTypes        #-}
{-# LANGUAGE DataKinds                  #-}
{-# LANGUAGE DeriveGeneric              #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}
{-# LANGUAGE OverloadedStrings          #-}
{-# LANGUAGE ScopedTypeVariables        #-}
{-# LANGUAGE TypeApplications           #-}
{-# LANGUAGE TypeFamilies               #-}
{-# LANGUAGE TypeOperators              #-}
module Businesstypes.ChatCommand where

import           Businesstypes.ChatRegistration
import           Businesstypes.ChatMessage
import           Data.Aeson
import           GHC.Generics

data ChatCommand = Register ChatRegistration | NewMessage ChatMessage deriving (Show, Eq, Generic)
instance ToJSON ChatCommand where
  toJSON (Register reg) =
    object [ "command" .= ("register"::String)
           , "registration" .= (toJSON reg)
           ]

  toJSON (NewMessage msg) =
    object [ "command" .= ("message"::String)
           , "chatMessage" .= (toJSON msg)
           ]
